modal/_container_io_manager.py

```diff
@@ -486,7 +485,10 @@ async def handle_input_exception(self, input_id, started_at: float) -> AsyncGene
         """Handle an exception while processing a function input."""
         try:
             yield
-        except KeyboardInterrupt:
+        except (KeyboardInterrupt, GeneratorExit):
```

Comment:
Fix 2: Reraise `GeneratorExit` instead of catching it further down and printing it/emitting a failed input